### ISSUE TITLE
FUPS ARE BREAKING THE GAME??? (FREE NUKE ANYTIME) (speedrunning has fallen...)

### ISSUE THREAD

**Issue by Giorgi3333** [Hypatech/ULTRAKILL-Speedrunner-Issues#13]
fups have been a very contentious thing for a while now, and most runners believe that they are overpowered. before i talk about any of the proposed changes im first going to lay out the reasons as to why we think its op.

fups give you a massive red explosion, at the cost of 35 HP, and no setup whatsoever, along with movement if you want. compare that to every other red explosion in the game, cores are wayy smaller and are harder to setup, sharp is super situational, nuking requires a railcharge. all of these options are balanced in some way, while fups are not. i feel like i really need to hammer home how fucking massive fup explosions are, they are like 70% the size of nukes. not to mention the fact that they can be stacked, if you whip 4 rockets to yourself at the same time you would output 28 damage, for reference an electric rail causes 8 damage. fups invalidate a lot of interesting routes and situations because the solution is just fup, which is a shame when you consider the diverse and complex ecosystem that is the ultrakill arsenal. having overpumps and core snipes invalidated in a bunch of spots because of it is sorta frustrating. 

understanding that fups are a problem is the easy part, the hard part is actually fixing them. after some discussion with runners we all generally agreed on the following changes:

1. either remove rocket stacking or make each rocket give diminishing returns (so if a fup deals 7 damage, each rocket added would deal 1.5 damage instead of the normal 7, so fupping 4 rockets at once would deal 11,5 damage instead of 28)
2. when fupping instead of a red explosion a yellow one should happen
3. fups deal 50 damage to the player

below is a video showing how insane rocket stacking is.

https://github.com/Hypatech/ULTRAKILL-Speedrunner-Issues/assets/167441401/55b7e450-142a-4e92-80c3-d588c3a5daae

**Comment by Hypatech** [comment]
looks good! what was the intent behind making fups not red btw? was it just for less range?

**Comment by Giorgi3333** [comment]
> looks good! what was the intent behind making fups not red btw? was it just for less range?

we actually had a lot of discussion around this and i thought about it a lot as well. red explosions have unique properties that make them better than yellow ones, most noteably completely bypassing the soldiers resistance to explosions. fups have the double wammy. as in, they have a massive radious AND are red explosions. we had to remove one or the other and we went with getting rid of the red explosion aspect.

**Comment by Hypatech** [comment]
> > looks good! what was the intent behind making fups not red btw? was it just for less range?
> 
> we actually had a lot of discussion around this and i thought about it a lot as well. red explosions have unique properties that make them better than yellow ones, most noteably completely bypassing the soldiers resistance to explosions. fups have the double wammy. as in, they have a massive radious AND are red explosions. we had to remove one or the other and we went with getting rid of the red explosion aspect.

@Giorgi3333 is making it yellow an arbitrary choice (where reducing the range would be perfectly good as a substitute) or was it determined to be a better choice?

**Comment by Hypatech** [comment]
Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better

**Comment by 10-days-till-xmas** [comment]
That could work, although u should keep it iframe-able how it was before, where you needed to dash twice in order to fully avoid the damage (a fair trade off, preventing spamming it)
Another option could be that fups can cause hard damage? Though then again most top players are constantly at high style, so this may be an issue (unless fups can bring down your style tier but that'll just cause an inconvenience). We could ask a high-level cybergrind player for their ideas too

**Comment by Giorgi3333** [comment]
> > > looks good! what was the intent behind making fups not red btw? was it just for less range?
> > 
> > 
> > we actually had a lot of discussion around this and i thought about it a lot as well. red explosions have unique properties that make them better than yellow ones, most noteably completely bypassing the soldiers resistance to explosions. fups have the double wammy. as in, they have a massive radious AND are red explosions. we had to remove one or the other and we went with getting rid of the red explosion aspect.
> 
> @Giorgi3333 is making it yellow an arbitrary choice (where reducing the range would be perfectly good as a substitute) or was it determined to be a better choice?

we did think that it was the best choice but honestly there are so many ways of going about nerfing fups that there could easily be another solution

**Comment by Giorgi3333** [comment]
> Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better

that could maybe work. so ur suggesting that it should stay a red explosion but make it smaller than an overpump?

**Comment by Hypatech** [comment]
yeah exactly

> > Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better
> 
> that could maybe work. so ur suggesting that it should stay a red explosion but make it smaller than an overpump?

yeah exactly! non iframeable too so overpump still has benefits

**Comment by Hypatech** [comment]
> That could work, although u should keep it iframe-able how it was before, where you needed to dash twice in order to fully avoid the damage (a fair trade off, preventing spamming it) Another option could be that fups can cause hard damage? Though then again most top players are constantly at high style, so this may be an issue (unless fups can bring down your style tier but that'll just cause an inconvenience). We could ask a high-level cybergrind player for their ideas too

Itd be hard to implement a double-dash on such a small radius explosion, prob more effort than its worth

**Comment by Giorgi3333** [comment]
> > That could work, although u should keep it iframe-able how it was before, where you needed to dash twice in order to fully avoid the damage (a fair trade off, preventing spamming it) Another option could be that fups can cause hard damage? Though then again most top players are constantly at high style, so this may be an issue (unless fups can bring down your style tier but that'll just cause an inconvenience). We could ask a high-level cybergrind player for their ideas too
> 
> Itd be hard to implement a double-dash on such a small radius explosion, prob more effort than its worth

i like the idea of it not being I frame-able

**Comment by Giorgi3333** [comment]
> yeah exactly
> 
> > > Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better
> > 
> > 
> > that could maybe work. so ur suggesting that it should stay a red explosion but make it smaller than an overpump?
> 
> yeah exactly! non iframeable too so overpump still has benefits

ehh, i dont hate it but i feel like the same problem would persist and il explain why. the tech that fups overshadow the most are core snipes. when fups got added sooo many core snipes just got replaced with fups. cores take a longer time to setup and they have a smaller radious. if you implement this change the problem of them overshadowing core snipes will probably still remain? since making fups have a smaller explosion size than core snipes seems ridiculous but at the same time anything less than that would still make it over shadow core snipes. fups are literally instant, as rockets already have the 2nd fastest draw time out of any weapon, meanwhile with core snipes you shoot the core slower and then you have to switch to the revolver, this takes a long ass time comparatively. all red explosions are not easy to setup (if you ignore fups), cores are slow like i just said and dont have a particularly big radious, sharp is super situational, and nukes need a rail charge. those sources of red explosions are all balanced, except fups.

**Comment by okEanra** [comment]
> Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better

i feel making its radius smaller than overpump would remove any use from fupping bcs with the other proposals it would be a smaller, less damaging to enemies, and undodgable yellow explosion compared to overpump

i agree with them being a yellow explosion so it wouldnt bypass soldiers/do as much damage, but reducing their size by that much seems a bit too much to me. (maybe a bit smaller than srs snipe size would work?)

**Comment by Giorgi3333** [comment]
> > Because we're a little iffy on making it a yellow explosion for regular players' sake, but if we were to make it say, smaller than an overpump (but not iframe-able!!!) that would be better
> 
> i feel making its radius smaller than overpump would remove any use from fupping bcs with the other proposals it would be a smaller, less damaging to enemies, and undodgable yellow explosion compared to overpump
> 
> i agree with them being a yellow explosion so it wouldnt bypass soldiers/do as much damage, but reducing their size by that much seems a bit too much to me. (maybe a bit smaller than srs snipe size would work?)

i dont think it would make fups irrelevant. the 2 biggest strenght of fups would remain, the fact that it doubles as a mobility tool, and the fact that its near instant. those 2 things stay, but it size gets greatly nerfed to not make a better core snipe.

**Comment by Hypatech** [comment]
To be clear:
- Red explosion
- (Slightly) smaller than overpump
- 35->50 damage (7->10 damage to enemies)
- Un iframeable

**Comment by Giorgi3333** [comment]
> To be clear:
> 
> * Red explosion
> * (Slightly) smaller than overpump
> * 35->50 damage (7->10 damage to enemies)
> * Un iframeable

oh wow i forgot about the 7 -> 10 damage part, is there not a way to keep the damage the same to enemies but increase it for v1?

**Comment by Hypatech** [comment]
Talked about it with hakita, he doesnt want to do that. In my eyes its offset by the fact you can only do it once before needing to heal + the smaller radius

**Comment by Giorgi3333** [comment]
> Talked about it with hakita, he doesnt want to do that. In my eyes its offset by the fact you can only do it once before needing to heal + the smaller radius

thats fair, my slight worry is that at the highest level of play this wouldnt change too much since we can almost always find a way to heal, but i could be wrong. frankly we have almost never had to deal with the issue of not having enough health but you could say that its because you could always dash away and it dealt 35 damage so yeah overall all of these changes look great.

**Comment by 10-days-till-xmas** [comment]
I think it should be a core snipe sized yellow explosion that deals 5 dmg (50 to the player) and would be iframeable as a result, but with fup stacking removed as well. We never iframe fups anyways so making it undodgeable doesn't make a difference. This would be as balanced as an overpump since in roughly the same time you can charge an overpump and explode, you could fup twice, and yield the same dps

**Comment by Giorgi3333** [comment]
> I think it should be a core snipe sized yellow explosion that deals 5 dmg (50 to the player) and would be iframeable as a result, but with fup stacking removed as well. We never iframe fups anyways so making it undodgeable doesn't make a difference. This would be as balanced as an overpump since in roughly the same time you can charge an overpump and explode, you could fup twice, and yield the same dps

well we didnt use to I frame fups because they did 35 damage, not 50. hakita alr said its not gonna become a yellow explosion for the casual players sake, which is completely understandable. also 5 damage is wayyy too low.

**Comment by 10-days-till-xmas** [comment]
Hmm maybe for now we can keep fups as they are and remove rocket stacking, and if we do need a nerf now, we can reduce the radius a bit, but not too much. Nerfing conductor and the alt shotgun should be done first, since we're getting a bit of power creep

**Comment by Giorgi3333** [comment]
> Hmm maybe for now we can keep fups as they are and remove rocket stacking, and if we do need a nerf now, we can reduce the radius a bit, but not too much. Nerfing conductor and the alt shotgun should be done first, since we're getting a bit of power creep

nah im completely ready for fups to get all the suggested nerfs, they are fucking busted to shit and just because other stuff is also busted doesnt mean that fups shouldnt be taken care of. btw whats wrong with conductor or the alt shotgun? if you can please make an issue and tag me in it

**Comment by 10-days-till-xmas** [comment]
> > Hmm maybe for now we can keep fups as they are and remove rocket stacking, and if we do need a nerf now, we can reduce the radius a bit, but not too much. Nerfing conductor and the alt shotgun should be done first, since we're getting a bit of power creep
> 
> 
> 
> nah im completely ready for fups to get all the suggested nerfs, they are fucking busted to shit and just because other stuff is also busted doesnt mean that fups shouldnt be taken care of. btw whats wrong with conductor or the alt shotgun? if you can please make an issue and tag me in it

I'm just worried that fups dealing 10 dmg will be more of a buff (they can insta virtues) and if fup stacking remains, only 2 rockets will be needed to kill a cerb, or almost kill a gutterman/tank. Health is rarely an issue for us so it wouldn't do anything other than be a mild inconvenience. A smaller radius would be good tho (either the same size as overpump or a bit smaller would be good)

**Comment by Giorgi3333** [comment]
> > > Hmm maybe for now we can keep fups as they are and remove rocket stacking, and if we do need a nerf now, we can reduce the radius a bit, but not too much. Nerfing conductor and the alt shotgun should be done first, since we're getting a bit of power creep
> > 
> > 
> > nah im completely ready for fups to get all the suggested nerfs, they are fucking busted to shit and just because other stuff is also busted doesnt mean that fups shouldnt be taken care of. btw whats wrong with conductor or the alt shotgun? if you can please make an issue and tag me in it
> 
> I'm just worried that fups dealing 10 dmg will be more of a buff (they can insta virtues) and if fup stacking remains, only 2 rockets will be needed to kill a cerb, or almost kill a gutterman/tank. Health is rarely an issue for us so it wouldn't do anything other than be a mild inconvenience. A smaller radius would be good tho (either the same size as overpump or a bit smaller would be good)

yeah the 10 damage is a bit nuts but i wanna play around with it a bit still, if its a massive problem then we could ask them to nerf it

**Comment by Giorgi3333** [comment]
honestly the biggest problem is that you can still heal off of the things you kill with fup, this is me trying to simulate what it would be like if the fup changes came about @Hypatech maybe make it so that you cant heal off of fups or something? :P 

https://github.com/Hypatech/ULTRAKILL-Speedrunner-Issues/assets/167441401/cd061a81-3438-4932-be40-d0f3de8de9d6

**Comment by sargesvrge** [comment]
I definitely think that rocket stacking needs to be super toned down and the healing needs to be reduced (maybe being cut in half?) but I think buffing the damage to yourself isn't really a good idea if the healing is reduced. 
Also, fups ARE faster than frame 1 ssjs and overpumps by a little bit, which is another issue I remembered but didn't recall being brought up here (data below collected by eanra).
Fups in the Cyber Grind are actually pretty balanced, and they're trade offs with Overpumps for less damage to everything but a bigger radius, at the cost of self damage and slightly more damage to (most) enemies. It's definitely an issue trying to balance them for speedruns vs CG since they're fine in CG but dominate in speedruns. As a CG player who uses them often, they'd be rendered much less effective in CG if a lot of the propsed nerfs are set into place.
For now, I think fup stacking definitely needs to go (it only impacts speedrunning iirc), the speed you gain from doing a fup needs to be nerfed, and maybe tweak the explosion healing and explosion radius, but keep everything else the same. I think it should be gradual and we should get rid of the super glaring issues first before getting the minor tweaks out of the way.

(EDIT): eanra said later that all of the data in the image below included vertical speed.

<img width="197" alt="fupdataeanra" src="https://github.com/Hypatech/ULTRAKILL-Speedrunner-Issues/assets/167507088/2270240b-04bf-4774-b3b6-4650bfdf2600">

**Comment by Hypatech** [comment]
@sargesvrge super useful points thank you! yeah with CG in mind it also makes me more cautious about reducing the range compared to before. i wasnt planning on touching the healing, but how do we feel about increased self damage + a speed reduction for now?

**Comment by 10-days-till-xmas** [comment]
> @sargesvrge super useful points thank you! yeah with CG in mind it also makes me more cautious about reducing the range compared to before. i wasnt planning on touching the healing, but how do we feel about increased self damage + a speed reduction for now?

I'm still cautious about the self-damage increase, especially in the context of cybergrind. However a speed reduction would definitely be a good change, since they're the only one out of 3 feasible ways to move in midair other than rocket riding, and they usually replace overpumps due to how fups don't need to charge

**Comment by Hypatech** [comment]
Definitely killing stacking

**Comment by Hypatech** [comment]
> > @sargesvrge super useful points thank you! yeah with CG in mind it also makes me more cautious about reducing the range compared to before. i wasnt planning on touching the healing, but how do we feel about increased self damage + a speed reduction for now?
> 
> I'm still cautious about the self-damage increase, especially in the context of cybergrind. However a speed reduction would definitely be a good change, since they're the only one out of 3 feasible ways to move in midair other than rocket riding, and they usually replace overpumps due to how fups don't need to charge

If I can get away without a self damage increase i will be happier because its a pain to implement

**Comment by Hypatech** [comment]
I also just found out that fups are more consistent in cardinal directions. Fuck!!! Fixed that

**Comment by sargesvrge** [comment]
> @sargesvrge super useful points thank you! yeah with CG in mind it also makes me more cautious about reducing the range compared to before. i wasnt planning on touching the healing, but how do we feel about increased self damage + a speed reduction for now?

If we're talking about CG, the self damage is kind of negligible because of the healing or if you can parry something/Guard Break a gutterman/find a way to heal immediately. I feel like 50 damage is a LOT but the healing from the explosion and whatever you can grab a heal off of might negate that too. Maybe somewhere between 35-50? Idk.
Testing around with reduced healing but different values of self damage would be cool, but I don't have UnityExplorer or a program like that to test, nor am I good with code.
Speedrunning is probably a different story which I have no experience of so I'm staying away from that.

**Comment by sargesvrge** [comment]
Tested a little more and it appears at very close proximity to enemies, the explosion heals 30 damage, while you deal 35 to yourself, resulting in a net hp loss of 5. I think healing might be worthwhile to look into, but an outright healblock would be bad (healblock was notorious for being terrible in CG at any patch earlier than 15a). 

Also, this heals the same amount on enemies WITHOUT rocket launcher vulnerability, like Swordsmachines. I tested and got the same healing results.


https://github.com/Hypatech/ULTRAKILL-Speedrunner-Issues/assets/167507088/aad0a549-3773-4bfe-8a82-11253ac6afda

**Comment by 10-days-till-xmas** [comment]
Changes involving healing from blood generated by explosions (that damaged the player) could fix this, I think heck may have mentioned it before?

**Comment by Hypatech** [comment]
Current plan I think is just to nerf fups to 60u/s, feedback seems positive on that one. I think that should eliminate a lot of the issues in runs, but i will make further adjustments in future. Just wanna take it a small bit at a time
